src/gpuarray_buffer_cuda.c

```diff
@@ -637,6 +643,7 @@ static int cuda_records(gpudata *a, int flags, CUstream s) {
   if (flags & CUDA_WAIT_WRITE)
     a->ctx->err = cuEventRecord(a->wev, s);
   cuda_exit(a->ctx);
+  a->ls = s;
```

Comment:
this is changed even if no flags was set. Should we return an error if no flag is set?

I'm not sure the shortcut logic is always valid. Is it fine to record more events then wait? I think so, but I'm not 100% sure.

I'm not sure the logic is good. I'll try to call you later this afternoon to discuss this.